src/settings/Models/LoanPolicy/LoanPolicy.test.js

```diff
@@ -0,0 +1,322 @@
+import LoanPolicy from './LoanPolicy';
+import RenewalsPolicy from './RenewalsPolicy';
+import RequestManagement from './RequestManagement';
+import LoansPolicy from './LoansPolicy';
+import { Metadata } from '../common';
+import {
+  intervalIdsMap,
+  loanProfileMap,
+  BEGINNING_OF_THE_NEXT_OPEN_SERVICE_POINT_HOURS,
+} from '../../../constants';
+
+describe('LoanPolicy', () => {
+  const basicPolicy = {
+    id: 'id',
+    name: 'name',
+    description: 'description',
+    loanable: true,
+    renewable: true,
+  };
+
+  const fullPolicy = {
+    ...basicPolicy,
+    loansPolicy: {},
+    renewalsPolicy: {
+      numberAllowed: 5,
+      period: {
+        intervalId: 'id',
+        duration: 0,
+      },
+    },
+    requestManagement: {},
+    metadata: {},
+  };
+
+  const loanPolicy = new LoanPolicy(fullPolicy);
+
+  it('"LoanPolicy" should have correct properties', () => {
+    expect(loanPolicy).toEqual(expect.objectContaining(basicPolicy));
+  });
+
+  it('"loansPolicy" property should be instance of "LoansPolicy"', () => {
+    expect(loanPolicy.loansPolicy).toBeInstanceOf(LoansPolicy);
+  });
+
+  it('"requestManagement" property should be instance of "RequestManagement"', () => {
+    expect(loanPolicy.requestManagement).toBeInstanceOf(RequestManagement);
+  });
+
+  it('"renewalsPolicy" property should be instance of "RenewalsPolicy"', () => {
+    expect(loanPolicy.renewalsPolicy).toBeInstanceOf(RenewalsPolicy);
+  });
+
+  it('"metadata" property should be instance of "Metadata"', () => {
+    expect(loanPolicy.metadata).toBeInstanceOf(Metadata);
+  });
+
+  it('"LoanPolicy" should have correct properties when "policy" is not provided', () => {
+    const loanPolicyInstance = new LoanPolicy();
+    const expectedResult = {
+      id: undefined,
+      name: undefined,
+      description: undefined,
+      loanable: false,
+      renewable: false,
+    };
+
+    expect(loanPolicyInstance).toEqual(expect.objectContaining(expectedResult));
+  });
+
+  it('"defaultLoanPolicy" should return correct data', () => {
+    expect(LoanPolicy.defaultLoanPolicy()).toEqual({
+      loanable: true,
+      renewable: true,
+    });
+  });
+
+  it('"hasValue" should return true', () => {
```

Comment:
Can we add more details to know why there should be true or false? Check please tests below too.